SU2_CFD/src/solvers/CIncNSSolver.cpp

```diff
@@ -366,3 +378,304 @@ void CIncNSSolver::BC_ConjugateHeat_Interface(CGeometry *geometry, CSolver **sol
     nodes->SetEnergy_ResTruncError_Zero(iPoint);
   }
 }
+
+
+
+void CIncNSSolver::SetTauWall_WF(CGeometry *geometry, CSolver **solver_container, CConfig *config) {
+
+  /*---  The wall function implemented herein is based on Nichols and Nelson AIAAJ v32 n6 2004.
+   At this moment, the wall function is only available for adiabatic flows.
+   ---*/
+  su2double RefDensity,RefVel2; 
+  if ((config->GetRef_Inc_NonDim() == DIMENSIONAL) || (config->GetRef_Inc_NonDim() == INITIAL_VALUES)) {
+    RefDensity = Density_Inf;
+    RefVel2 = 0.0;
+    for (auto iDim = 0u; iDim < nDim; iDim++) 
+      RefVel2 += Velocity_Inf[iDim] * Velocity_Inf[iDim];
+  } else if (config->GetRef_Inc_NonDim() == REFERENCE_VALUES) {
+    RefDensity = config->GetInc_Density_Ref();
+    RefVel2 = config->GetInc_Velocity_Ref() * config->GetInc_Velocity_Ref();
+  }
+
+  bool debug = true;
+  unsigned long counter;
+
+  su2double UnitNormal[3];
+  su2double **grad_primvar, tau[3][3];
```

Comment:
Use [MAXNDIM] instead of [3] please. It is not so much that we want to run 4D one day, but MAXNDIM and MAXNVAR make it unambiguous whether the variable is related with the geometry or the solution.
Also make grad_primvar `const su2double* const*` for peace of mind.

Reply:
Thanks, it's a habit that's hard to break...